### ISSUE TITLE
Inbound Filters by tags

### ISSUE THREAD

**Issue by keradus** [getsentry/sentry#22874]
hey, it would be great to have a way to set up Inbound Filters based on value of tags - eg to drop all incoming events from given user (I know it's doable now with IP, yet many users can share single IP)

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-2339) by [Unito](https://www.unito.io)

**Comment by BYK** [comment]
@markstory is this being considered (so we should add "Status: On Hold") or we are unlikely to attend?

**Comment by keradus** [comment]
I would appreciate your decision whether the feature request makes sense ;)

**Comment by shadow-cs** [comment]
I'd find that extremely useful indeed. I'd like to filter devices (we use Sentry in an IoT world) based on a serial number if one goes haywire (since the devices are usually behind cell operator gateway, IP filtering won't do us much good in this case).

**Comment by markstory** [comment]
I'm not overly close to the roadmap for inbound filters. There is a team working on evolving inbound filters but tags are not currently part of that scope either.

**Comment by shadow-cs** [comment]
@markstory is there an epic or some docs we can read regarding the evolution of the inbound filters?

**Comment by jan-auer** [comment]
@shadow-cs We can't share a public roadmap on inbound filters just yet. Most likely, I can share something in summer. 
Controlling traffic by tags is definitely on our radar.

**Comment by keradus** [comment]
thanks for the news, @jan-auer !

**Comment by watzon** [comment]
I agree that this would make a fantastic addition. Would PRs be welcome? I might be willing to get my hands dirty if it could speed things up.

**Comment by parichay28** [comment]
+1. This would indeed be a handy feature 🙌

**Comment by jlaz-globality** [comment]
This would also be very handy for us!

**Comment by mhuebner** [comment]
+1 very important feature

**Comment by jarrodlombardo-EventBase** [comment]
We have some clients hang on to old versions of our apps for many months and not being able to filter by tags eats up 100-200k events from our quota every month. Please implement this.

**Comment by cytivrat** [comment]
Any chance this feature will be implemented in foreseen future?

**Comment by Edgar-Cordova** [comment]
This would be useful for my product too.

**Comment by azaslavsky** [comment]
Moved this to Issues, though maybe Settings - Relay is a better home?

**Comment by JeremiahDo** [comment]
I would also very much appreciate this. It would help to filter out all errors for a certain env during a maintenance.

**Comment by tdoyle01** [comment]
+1, would love to have this feature! We have test servers that we spin up sometimes and it'd be great to have the ability to filter out events from those before they hit the Sentry inbox (we currently can only identify them using tags).

**Comment by ellenblaine** [comment]
This would be a great feature! For my use case, I have some events coming from a library that I don't have much control over (thus can't filter in my user level code), and only the tag has meaningful content for filtering (the title is pretty generic).

**Comment by ubifred** [comment]
+1

**Comment by ericriff** [comment]
I'm also interested on this.

**Comment by rachrwang** [comment]
Hi everyone - thank you for sharing your interest in this ticket. We don't have immediate plans to work on this, though it's very helpful for future prioritization discussions.

**Comment by keradus** [comment]
very unfortunate msg after 4 years of waiting

**Comment by keradus** [comment]
@Angelodaniel , would that help with preventing those events to be consumed at all? or only remove already consumed events? (but allow them to be reported again)

remember I want to filter them out in a way like they were never sent and that they do not consume any quota/limits
> Inbound Data Filters
> Filters allow you to prevent Sentry from storing events in certain situations. Filtered events are tracked separately from rate limits, and do not apply to any project quotas.

**Comment by lobsterkatie** [comment]
Delete and discard means you never see them again. Keep in mind it's only available on the Business plan, though.

**Comment by Angelodaniel** [comment]
@keradus it will basically put a Issue (fingerprint value) to the inbound filters. So no they won't be accepted, and you won't be able to look into them obviously.

**Comment by dajju** [comment]
This feature would be great!
